device/core/test/_internal_client_test.js

```diff
@@ -18,441 +16,335 @@ var Message = require('azure-iot-common').Message;
-        }, ReferenceError);
+      /*Tests_SRS_NODE_INTERNAL_CLIENT_16_024: [The ‘setTransportOptions’ method shall throw a ‘ReferenceError’ if the options object is falsy] */
+      [null, undefined, '', 0].forEach(function (option) {
+        it('throws a ReferenceError if options is ' + option, function () {
+          var client = new ClientCtor(new FakeTransport());
+          assert.throws(function () {
+            client.setTransportOptions(option, function () { });
+          }, ReferenceError);
+        });
       });
-    });
-
-    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_025: [The ‘setTransportOptions’ method shall throw a ‘NotImplementedError’ if the transport doesn’t implement a ‘setOption’ method.]*/
-    it('throws a NotImplementedError if the setOptions method is not implemented on the transport', function () {
-      var client = new InternalClient(new EventEmitter());
 
-      assert.throws(function () {
-        client.setTransportOptions({ foo: 42 }, function () { });
-      }, errors.NotImplementedError);
-    });
-  });
+      /*Tests_SRS_NODE_INTERNAL_CLIENT_16_025: [The ‘setTransportOptions’ method shall throw a ‘NotImplementedError’ if the transport doesn’t implement a ‘setOption’ method.]*/
+      it('throws a NotImplementedError if the setOptions method is not implemented on the transport', function () {
+        var client = new ClientCtor(new EventEmitter());
 
-  describe('#setOptions', function() {
-    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_042: [The `setOptions` method shall throw a `ReferenceError` if the options object is falsy.]*/
-    [null, undefined].forEach(function(options) {
-      it('throws is options is ' + options, function() {
-        var client = new InternalClient(new EventEmitter());
         assert.throws(function () {
-          client.setOptions(options, function () { });
-        }, ReferenceError);
+          client.setTransportOptions({ foo: 42 }, function () { });
+        }, errors.NotImplementedError);
       });
     });
 
-    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_043: [The `done` callback shall be invoked no parameters when it has successfully finished setting the client and/or transport options.]*/
-    it('calls the done callback with no parameters when it has successfully configured the transport', function(done) {
-      var client = new InternalClient(new FakeTransport());
-      client.setOptions({}, done);
-    });
-
-    /*Tests_SRS_NODE_INTERNAL_CLIENT_16_044: [The `done` callback shall be invoked with a standard javascript `Error` object and no result object if the client could not be configured as requested.]*/
-    it('calls the done callback with an error when it failed to configured the transport', function(done) {
-      var failingTransport = new FakeTransport();
-      sinon.stub(failingTransport, 'setOptions').callsFake(function (options, done) {
-        done(new Error('dummy error'));
+    describe('#setOptions', function () {
```

Comment:
>describe('#setOptions', function () { [](start = 4, length = 37)

nit: adding this space seems arbitrary.  Should we do a one-time prettifier pass?